protocols/mempool/protocol-abstract.md

```diff
@@ -0,0 +1,232 @@
+            send(<TX, tx>) to q
+
+upon commit(txs):
+    for all tx in txs
+        seen[tx] = true
+```
+
+The algorithm represents the operation of a node in the network.
+The `peers` variable stores the list of other nodes to which the node is
+directly connected to in the overlay network.
+The `peers` list is dynamic, being updated by the underlying p2p communication
+layer when new connections are established or existing connections are dropped.
+
+Transactions are disseminated by flooding, as the node sends every transaction
+it receives for the first time to all its `peers`.
+A transaction can be received from an external component, via the `brodcast`
+method, or from a peer, via the `receive` method.
+In the case a transaction is received from a peer, the node does not send the
+same transaction back to that peer.
+
+To determine whether a transaction is received for the first time, the algorithm
+maintains a map `seen` containing every transaction ever received.
+This map prevents transactions from being indefinitely forwarded to the peers
+and provides a simple mechanism to cease the propagation of transactions when
+they are received by every node in the network.
+
+The algorithm also uses the `seen` map to mark transactions that have been
+committed to the blockchain, and therefore do not need to be further
+propagated.
+
+### Limitations
+
+While useful to represent the operation of the transaction propagation
+protocol, the above algorithm has some practical limitations:
+
+1. New peers: the algorithm does not send transactions whose dissemination is
+   ongoing to newly connected peers.
+In other words, a node joining the network should only expected to receive
+transactions whose dissemination has started after it has joined the network.
+2. Ordering: the algorithm may not send transactions that are concurrently
+   being disseminated in the same order to  peers.
+In fact, in case of concurrent invocations of the upon broadcast or receive
+clauses of the algorithm, peers are likely to receive the relayed transactions
+in different orders.
+
+The next algorithm is refinement of the above presented algorithm so that to
+address those limitations.
+This first algorithm, however, is relevant because it is easier to analyze
+and allows a simplified representation of the 
+[propagation mechanism](./protocol-abstract.md#analysis-of-the-propagation).
+
+> It is worth noting that the mempool does not (and cannot) provide any guarantee
+> of transaction ordering when sending transactions to peers (either existing or new).
+> However, users of the protocol expect the mempool protocol to address the above
+> two limitations, which can be attested by the only document specifically
+> addressing the mempool in CometBFT documentation:
+> https://github.com/cometbft/cometbft/blob/main/docs/core/mempool.md
+
+
+## Mempool flooding algorithm
+
+The following algorithm is obtained by refining the previous algorithm as follows.
```

Comment:
"following" is a bit odd here 

In fact, the code below takes into account the mempool itself. So, strictly speaking, this is more than a refinment.